src/types/pathways-model.d.ts

```diff
@@ -30,17 +30,19 @@ declare module 'pathways-model' {
     key?: string;
     label: string;
     transitions: Transition[];
+    nodeTypeIsUndefined?: boolean;
   }
 
   export interface GuidanceState extends State {
     cql: string;
     action: Action[];
   }
 
-  // NOTE: the model also includes a BranchState (which extends State),
-  // but as of right now it has no additional fields not in State,
-  // and TypeScript does not allow "empty" interfaces so we can't add it yet.
-  // Add it here if/when we ever need it.
+  export interface BranchState extends State {
+    criteriaSource?: string;
+    mcodeCriteria?: string;
+    otherCriteria?: string;
```

Comment:
Do we need both of these? Isn't it implied whether or not it is mcode criteria based on the source?

Reply:
Both of these will be removed in a future task when we implement branch criteria.